charts/karpenter/values.yaml

```diff
@@ -101,11 +101,26 @@ extraObjects: []
 #  kind: AWSNodeTemplate
 #  metadata:
 #    name: default
+#    annotations:
+#      "helm.sh/hook": "post-install, post-upgrade"
+#      "helm.sh/hook-weight": "1"
+#      "helm.sh/hook-delete-policy": before-hook-creation
 #  spec:
 #    subnetSelector:
 #      karpenter.sh/discovery: {CLUSTER_NAME}
 #    securityGroupSelector:
 #      karpenter.sh/discovery: {CLUSTER_NAME}
+#- apiVersion: karpenter.sh/v1alpha5
+#  kind: Provisioner
+#  metadata:
+#    name: default
+#    annotations:
+#      "helm.sh/hook": "post-install, post-upgrade"
+#      "helm.sh/hook-weight": "5"
```

Comment:
```suggestion
#      "helm.sh/hook-weight": "1"
```

Any reason this has a different weight?

Reply:
Since the provisioner relies on the awsnodetemplate it made more sense for me to create the template first, and if it fails, to not continue.
That's more an OCD-compliant setting than a real issue, if you think everything should be applied at once that's fine too.

Reply:
I'd remove the ordering. If we can't find the node template for the provisioner we'll just ignore the provisioner until the node template gets applied